resources/services/backup/plans.go

```diff
@@ -0,0 +1,383 @@
+					},
+					{
+						Name:        "resources",
+						Description: "A list of Amazon Resource Names (ARNs) to assign to a backup plan.",
+						Type:        schema.TypeStringArray,
+						Resolver:    schema.PathResolver("BackupSelection.Resources"),
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchBackupPlans(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListBackupPlansInput{MaxResults: aws.Int32(100)}
+	for {
+		result, err := svc.ListBackupPlans(ctx, &params, func(o *backup.Options) {
+			o.Region = cl.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, m := range result.BackupPlansList {
+			plan, err := svc.GetBackupPlan(
+				ctx,
+				&backup.GetBackupPlanInput{BackupPlanId: m.BackupPlanId, VersionId: m.VersionId},
+				func(o *backup.Options) { o.Region = cl.Region },
+			)
+			if err != nil {
+				return diag.WrapError(err)
+			}
+			if plan != nil {
+				res <- *plan
+			}
+		}
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		params.NextToken = result.NextToken
+	}
+	return nil
+}
+
+func resolvePlanAdvancedBackupSettings(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	plan := resource.Item.(backup.GetBackupPlanOutput)
+	b, err := json.Marshal(plan.AdvancedBackupSettings)
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set(c.Name, b))
+}
+
+func resolvePlanTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	plan := resource.Item.(backup.GetBackupPlanOutput)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListTagsInput{ResourceArn: plan.BackupPlanArn}
+	tags := make(map[string]string)
+	for {
+		result, err := svc.ListTags(ctx, &params, func(o *backup.Options) { o.Region = cl.Region })
+		if result == nil {
+			break
+		}
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for k, v := range result.Tags {
+			tags[k] = v
+		}
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		params.NextToken = result.NextToken
+	}
+	return diag.WrapError(resource.Set(c.Name, tags))
+}
+
+func fetchBackupSelections(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	plan := parent.Item.(backup.GetBackupPlanOutput)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListBackupSelectionsInput{
+		BackupPlanId: plan.BackupPlanId,
+		MaxResults:   aws.Int32(100),
```

Comment:
Is this the max that the endpoint supports?

Reply:
SDK does not specify any bounds. Some services are fine without the limit (using some default value), some require it explicitly, so I just picked 100 as a "good enough" value.

Reply:
[The max is 1000](https://docs.aws.amazon.com/aws-backup/latest/devguide/API_ListBackupSelections.html), Can we set it to that? I would rather try and minimize API requests

Reply:
Great, I'll set it to 1000.